### ISSUE TITLE
Rule proposal: `no-useless-iterator-to-array`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#2830]
### Description

Very similar to [no-useless-spread](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-useless-spread.md), but I don't think it make sense to add to `no-useless-spread`

### Examples

```js
// ❌
const set = new Set(iterator.toArray());

// ✅
const set = new Set(iterator);
```

```js
// ❌
for (const foo of iterator.toArray());

// ✅
for (const foo of iterator);
```

More examples see [no-useless-spread](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-useless-spread.md), replace `...` to `.toArray()`

### Proposed rule name

no-useless-iterator-to-array

**Comment by fisker** [comment]
Should also report array methods that also exists in `Iterator`

```js
// ❌
iterator.toArray().every()

// ✅
iterator.every()
```

```js
// ❌
iterator.toArray().find()

// ✅
iterator.find()
```

Full list https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator

<img width="978" height="518" alt="Image" src="https://github.com/user-attachments/assets/66a91a20-20f0-428a-9691-1ab4f4ea2266" />

__`map` and `flatMap` should not include__

**Comment by silverwind** [comment]
How would the rule detect whether something is an iterator? Does it do that via type information to detect the presence of a `.next` method?

**Comment by fisker** [comment]
This rule is about `.toArray()`. Not `new Set(Array.from(iterable))`, since there is no way to know the argument of `Array.from()` is `Iterator`.

> Array.from(iterable) is a more well-known alternative

 `.toArray()` should be preferred when possible https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2567

**Comment by maifeeulasad** [comment]
Willing to work on this one. Can I get assigned to this?
